File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/Python/s081988820_comments_removed.py
import sys
from typing import List, Optional

class Main:
    def solve(self):
        H, W = map(int, input().split())
        c = [list(map(int, input().split())) for _ in range(10)]
        min_cost = [c[i][1] for i in range(10)]
        
        for _ in range(10):
            for i in range(10):
                for j in range(10):
                    min_cost[i] = min(min_cost[i], c[i][j] + min_cost[j])
        
        ans = 0
        for _ in range(H):
            A = list(map(int, input().split()))
            for a in A:
                if a >= 0:
                    ans += min_cost[a]
        
        print(ans)

class Scanner:
    def __init__(self, input_stream):
        self.in_stream = input_stream
        self.buffer = bytearray(1024)
        self.index = 0
        self.length = 0

    def is_printable_char(self, c):
        return '!' <= chr(c) <= '~'

    def is_digit(self, c):
        return '0' <= chr(c) <= '9'

    def has_next_byte(self):
        if self.index < self.length:
            return True
        try:
            self.length = self.in_stream.readinto(self.buffer)
            self.index = 0
        except Exception as e:
            print(e)
        return self.length > 0

    def has_next(self):
        while self.has_next_byte() and not self.is_printable_char(self.buffer[self.index]):
            self.index += 1
        return self.has_next_byte()

    def read_byte(self):
        return self.buffer[self.index] if self.has_next_byte() else -1

    def next(self):
        if not self.has_next():
            raise RuntimeError("no input")
        sb = []
        b = self.read_byte()
        while self.is_printable_char(b):
            sb.append(chr(b))
            b = self.read_byte()
        return ''.join(sb)

    def next_long(self):
        if not self.has_next():
            raise RuntimeError("no input")
        value = 0
        minus = False
        b = self.read_byte()
        if b == ord('-'):
            minus = True
            b = self.read_byte()
        while self.is_printable_char(b):
            if self.is_digit(b):
                value = value * 10 + (b - ord('0'))
            b = self.read_byte()
        return -value if minus else value

    def next_int(self):
        return int(self.next_long())

    def next_double(self):
        return float(self.next())

class CombCalculator:
    def comb(self, n: int, m: int) -> int:
        pass

class MobiusFunction:
    def get(self, n: int) -> int:
        pass

class SieveMobiusFunction(MobiusFunction):
    def __init__(self, size: int):
        self.size = size
        self.mobius_function_values = [0] * size
        self.mobius_function_values[1] = 1
        for i in range(2, size):
            self.mobius_function_values[i] = 1
        for i in range(2, int(size ** 0.5) + 1):
            for k in range(1, (size - 1) // (i * i) + 1):
                self.mobius_function_values[i * i * k] = 0
        for i in range(2, size):
            if self.mobius_function_values[i] == 1:
                for k in range(1, (size - 1) // i + 1):
                    self.mobius_function_values[i * k] *= -1
            if self.mobius_function_values[i] > 1:
                self.mobius_function_values[i] = 1
            if self.mobius_function_values[i] < -1:
                self.mobius_function_values[i] = -1

    def get(self, n: int) -> int:
        if n > self.size:
            raise RuntimeError("n is greater than size.")
        if n < 0:
            return 0
        return self.mobius_function_values[n]

class PrimeFactorizationMobiusFunction(MobiusFunction):
    def get(self, n: int) -> int:
        if n <= 0:
            return 0
        if n == 1:
            return 1
        num = 0
        for i in range(2, n):
            if n % i == 0:
                n //= i
                num += 1
                if n % i == 0:
                    return 0
        return -1 if num % 2 else 1

class FactorialTableCombCalculator(CombCalculator):
    def __init__(self, size: int, mod: int):
        self.size = size
        self.factorial_table = [1] * (size + 1)
        self.inverse_factorial_table = [1] * (size + 1)
        self.mod = mod
        for i in range(1, size + 1):
            self.factorial_table[i] = (self.factorial_table[i - 1] * i) % mod
        self.inverse_factorial_table[size] = self.inverse(self.factorial_table[size], mod)
        for i in range(size - 1, 0, -1):
            self.inverse_factorial_table[i] = (self.inverse_factorial_table[i + 1] * (i + 1)) % mod

    def inverse(self, n: int, mod: int) -> int:
        return pow(n, mod - 2, mod)

    def comb(self, n: int, m: int) -> int:
        if n > self.size:
            raise RuntimeError("n is greater than size.")
        if n < 0 or m < 0 or n < m:
            return 0
        return (((self.factorial_table[n] * self.inverse_factorial_table[m]) % self.mod) * self.inverse_factorial_table[n - m]) % self.mod

class TableCombCalculator(CombCalculator):
    def __init__(self, size: int, mod: int):
        self.size = size
        self.table = [[1] + [0] * i for i in range(size + 1)]
        for n in range(1, size + 1):
            for m in range(1, n):
                self.table[n][m] = (self.table[n - 1][m - 1] + self.table[n - 1][m]) % mod
            self.table[n][n] = 1

    def comb(self, n: int, m: int) -> int:
        if n > self.size:
            raise RuntimeError("n is greater than size.")
        if n < 0 or m < 0 or n < m:
            return 0
        return self.table[n][m]

class Graph:
    def link(self, from_: int, to: int, cost: int):
        pass

    def get_cost(self, from_: int, to: int) -> Optional[int]:
        pass

    def get_vertex_num(self) -> int:
        pass

class FlowResolver:
    def max_flow(self, from_: int, to: int) -> int:
        pass

class ArrayGraph(Graph):
    def __init__(self, n: int):
        self.cost_array = [[None] * n for _ in range(n)]
        self.vertex_num = n

    def link(self, from_: int, to: int, cost: int):
        self.cost_array[from_][to] = cost

    def get_cost(self, from_: int, to: int) -> Optional[int]:
        return self.cost_array[from_][to]

    def get_vertex_num(self) -> int:
        return self.vertex_num

class DfsFlowResolver(FlowResolver):
    def __init__(self, graph: Graph):
        self.graph = graph

    def max_flow(self, from_: int, to: int) -> int:
        sum_ = 0
        current_flow = 0
        while True:
            current_flow = self.flow(from_, to, sys.maxsize // 3, [False] * self.graph.get_vertex_num())
            sum_ += current_flow
            if current_flow == 0:
                break
        return sum_

    def flow(self, from_: int, to: int, current_flow: int, passed: List[bool]) -> int:
        passed[from_] = True
        if from_ == to:
            return current_flow
        for id_ in range(self.graph.get_vertex_num()):
            if passed[id_]:
                continue
            cost = self.graph.get_cost(from_, id_)
            if cost and cost > 0:
                next_flow = min(current_flow, cost)
                return_flow = self.flow(id_, to, next_flow, passed)
                if return_flow > 0:
                    self.graph.link(from_, id_, cost - return_flow)
                    self.graph.link(id_, from_, self.graph.get_cost(id_, from_) or 0 + return_flow)
                    return return_flow
        return 0

class BinaryIndexedTree:
    def __init__(self, size: int):
        self.array = [0] * (size + 1)

    def add(self, index: int, value: int):
        while index < len(self.array):
            self.array[index] += value
            index += index & -index

    def get_sum(self, index: int) -> int:
        sum_ = 0
        while index > 0:
            sum_ += self.array[index]
            index -= index & -index
        return sum_

class BinaryIndexedTree2D:
    def __init__(self, size1: int, size2: int):
        self.array = [[0] * (size2 + 1) for _ in range(size1 + 1)]

    def add(self, index1: int, index2: int, value: int):
        i1 = index1
        while i1 < len(self.array):
            i2 = index2
            while i2 < len(self.array[0]):
                self.array[i1][i2] += value
                i2 += i2 & -i2
            i1 += i1 & -i1

    def get_sum(self, index1: int, index2: int) -> int:
        sum_ = 0
        i1 = index1
        while i1 > 0:
            i2 = index2
            while i2 > 0:
                sum_ += self.array[i1][i2]
                i2 -= i2 & -i2
            i1 -= i1 & -i1
        return sum_

class UnionFind:
    def union(self, A: int, B: int):
        pass

    def judge(self, A: int, B: int) -> bool:
        pass

    def get_set(self, id_: int) -> List[int]:
        pass

class SetUnionFind(UnionFind):
    def __init__(self, size: int):
        self.parent = list(range(size))
        self.rank = [0] * size
        self.map_ = {i: {i} for i in range(size)}

    def union(self, A: int, B: int):
        rootA = self.root(A)
        rootB = self.root(B)
        if rootA != rootB:
            if self.rank[rootA] < self.rank[rootB]:
                self.union_to(rootA, rootB)
            else:
                self.union_to(rootB, rootA)
                if self.rank[rootA] == self.rank[rootB]:
                    self.rank[rootA] += 1

    def union_to(self, source: int, dest: int):
        self.parent[source] = dest
        self.map_[dest].update(self.map_[source])

    def judge(self, A: int, B: int) -> bool:
        return self.root(A) == self.root(B)

    def get_set(self, id_: int) -> List[int]:
        return list(self.map_[self.root(id_)])

    def root(self, id_: int) -> int:
        if self.parent[id_] == id_:
            return id_
        self.parent[id_] = self.root(self.parent[id_])
        return self.parent[id_]

class ArrayUnionFind(UnionFind):
    def __init__(self, size: int):
        self.parent = list(range(size))
        self.rank = [0] * size
        self.size = size

    def union(self, A: int, B: int):
        rootA = self.root(A)
        rootB = self.root(B)
        if rootA != rootB:
            if self.rank[rootA] < self.rank[rootB]:
                self.union_to(rootA, rootB)
            else:
                self.union_to(rootB, rootA)
                if self.rank[rootA] == self.rank[rootB]:
                    self.rank[rootA] += 1

    def union_to(self, source: int, dest: int):
        self.parent[source] = dest

    def judge(self, A: int, B: int) -> bool:
        return self.root(A) == self.root(B)

    def get_set(self, id_: int) -> List[int]:
        return [i for i in range(self.size) if self.judge(i, id_)]

    def root(self, id_: int) -> int:
        if self.parent[id_] == id_:
            return id_
        self.parent[id_] = self.root(self.parent[id_])
        return self.parent[id_]

class PrimeNumberUtils:
    def __init__(self, limit: int):
        if limit > 10000000:
            print("上限の値が高すぎるため素数ユーティリティの初期化でTLEする可能性が大変高いです")
        self.primes = []
        self.is_prime_array = [False] * limit
        if limit > 2:
            self.primes.append(2)
            self.is_prime_array[2] = True
        for i in range(3, limit, 2):
            if self.is_prime(i, self.primes):
                self.primes.append(i)
                self.is_prime_array[i] = True

    def get_prime_number_list(self) -> List[int]:
        return self.primes

    def is_prime(self, n: int) -> bool:
        return self.is_prime_array[n]

    def is_prime(self, n: int, primes: List[int]) -> bool:
        for prime in primes:
            if n % prime == 0:
                return False
            if prime > int(n ** 0.5):
                break
        return True

class BitSet:
    def set(self, index: int, bit: bool):
        pass

    def get(self, index: int) -> bool:
        pass

    def shift_right(self, num: int):
        pass

    def shift_left(self, num: int):
        pass

    def or_(self, bitset: 'BitSet'):
        pass

    def and_(self, bitset: 'BitSet'):
        pass

class LongBit(BitSet):
    def __init__(self, size: int):
        self.bit_array = [0] * ((size + 63) // 64)

    def set(self, index: int, bit: bool):
        segment = index // 64
        in_index = index % 64
        if bit:
            self.bit_array[segment] |= 1 << in_index
        else:
            self.bit_array[segment] &= ~(1 << in_index)

    def get(self, index: int) -> bool:
        segment = index // 64
        in_index = index % 64
        return (self.bit_array[segment] & (1 << in_index)) != 0

    def shift_right(self, num: int):
        shift_seg = num // 64
        shift_in_i = num % 64
        for segment in range(len(self.bit_array)):
            source_seg = segment + shift_seg
            if source_seg < len(self.bit_array):
                self.bit_array[segment] = self.bit_array[source_seg] >> shift_in_i
                if shift_in_i > 0 and source_seg + 1 < len(self.bit_array):
                    self.bit_array[segment] |= self.bit_array[source_seg + 1] << (64 - shift_in_i)
            else:
                self.bit_array[segment] = 0

    def shift_left(self, num: int):
        shift_seg = num // 64
        shift_in_i = num % 64
        for segment in range(len(self.bit_array) - 1, -1, -1):
            source_seg = segment - shift_seg
            if source_seg >= 0:
                self.bit_array[segment] = self.bit_array[source_seg] << shift_in_i
                if shift_in_i > 0 and source_seg > 0:
                    self.bit_array[segment] |= self.bit_array[source_seg - 1] >> (64 - shift_in_i)
            else:
                self.bit_array[segment] = 0

    def get_long(self, segment: int) -> int:
        return self.bit_array[segment]

    def or_(self, bitset: 'BitSet'):
        if not isinstance(bitset, LongBit):
            return
        for segment in range(len(self.bit_array)):
            self.bit_array[segment] |= bitset.get_long(segment)

    def and_(self, bitset: 'BitSet'):
        if not isinstance(bitset, LongBit):
            return
        for segment in range(len(self.bit_array)):
            self.bit_array[segment] &= bitset.get_long(segment)

if __name__ == "__main__":
    main = Main()
    main.solve()
# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/cot/Java/C++/s081988820_comments_removed.cpp
#include <iostream>
#include <vector>
#include <optional>
#include <cmath>
#include <unordered_map>
#include <set>
#include <algorithm>

class Main {
public:
    void main() {
        Main main;
        main.solve();
    }

private:
    void solve() {
        std::cin >> H >> W;
        std::vector<std::vector<int>> c(10, std::vector<int>(10));
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                std::cin >> c[i][j];
            }
        }
        std::vector<int> min(10);
        for (int i = 0; i < 10; i++) {
            min[i] = c[i][1];
        }
        for (int tc = 0; tc < 10; tc++) {
            for (int i = 0; i < 10; i++) {
                for (int j = 0; j < 10; j++) {
                    min[i] = std::min(min[i], c[i][j] + min[j]);
                }
            }
        }
        int ans = 0;
        for (int h = 1; h <= H; h++) {
            for (int w = 1; w <= W; w++) {
                int A;
                std::cin >> A;
                if (A >= 0) {
                    ans += min[A];
                }
            }
        }
        std::cout << ans << std::endl;
    }

    int H, W;
};

class SieveMobiusFunction {
public:
    SieveMobiusFunction(int size) : size(size), mobiusFunctionValues(size) {
        mobiusFunctionValues[0] = 0;
        mobiusFunctionValues[1] = 1;
        for (int i = 2; i < size; i++) {
            mobiusFunctionValues[i] = 1;
        }
        for (int i = 2; i * i < size; i++) {
            for (int k = 1; i * i * k < size; k++) {
                mobiusFunctionValues[i * i * k] *= 0;
            }
        }

        for (int i = 2; i < size; i++) {
            if (mobiusFunctionValues[i] == 1) {
                for (int k = 1; i * k < size; k++) {
                    mobiusFunctionValues[i * k] *= -2;
                }
            }
            if (mobiusFunctionValues[i] > 1) {
                mobiusFunctionValues[i] = 1;
            }
            if (mobiusFunctionValues[i] < -1) {
                mobiusFunctionValues[i] = -1;
            }
        }
    }

    int get(int n) {
        if (n > size) {
            throw std::runtime_error("n is greater than size.");
        }
        if (n < 0) {
            return 0;
        }
        return mobiusFunctionValues[n];
    }

private:
    int size;
    std::vector<int> mobiusFunctionValues;
};

class PrimeFactorizationMobiusFunction {
public:
    int get(int n) {
        if (n <= 0) {
            return 0;
        }
        if (n == 1) {
            return 1;
        }
        int num = 0;
        for (int i = 2; i < n; i++) {
            if (n % i == 0) {
                n /= i;
                num++;
                if (n % i == 0) {
                    return 0;
                }
            }
        }
        return num % 2 == 0 ? -1 : 1;
    }
};

class FactorialTableCombCalculator {
public:
    FactorialTableCombCalculator(int size, long mod) : size(size), mod(mod), factorialTable(size + 1), inverseFactorialTable(size + 1) {
        factorialTable[0] = 1L;
        for (int i = 1; i <= size; i++) {
            factorialTable[i] = (factorialTable[i - 1] * i) % mod;
        }
        inverseFactorialTable[size] = inverse(factorialTable[size], mod);
        for (int i = size - 1; i >= 0; i--) {
            inverseFactorialTable[i] = (inverseFactorialTable[i + 1] * (i + 1)) % mod;
        }
    }

    long comb(int n, int m) {
        if (n > size) {
            throw std::runtime_error("n is greater than size.");
        }
        if (n < 0 || m < 0 || n < m) {
            return 0L;
        }
        return (((factorialTable[n] * inverseFactorialTable[m]) % mod) * inverseFactorialTable[n - m]) % mod;
    }

private:
    int size;
    long mod;
    std::vector<long> factorialTable;
    std::vector<long> inverseFactorialTable;

    long inverse(long n, long mod) {
        return pow(n, mod - 2, mod);
    }

    long pow(long n, long p, long mod) {
        if (p == 0) {
            return 1L;
        }
        long half = pow(n, p / 2, mod);
        long ret = (half * half) % mod;
        if (p % 2 == 1) {
            ret = (ret * n) % mod;
        }
        return ret;
    }
};

class TableCombCalculator {
public:
    TableCombCalculator(int size, long mod) : size(size), table(size + 1) {
        table[0] = std::vector<long>(1, 1L);
        for (int n = 1; n <= size; n++) {
            table[n] = std::vector<long>(n + 1);
            table[n][0] = 1L;
            for (int m = 1; m < n; m++) {
                table[n][m] = (table[n - 1][m - 1] + table[n - 1][m]) % mod;
            }
            table[n][n] = 1L;
        }
    }

    long comb(int n, int m) {
        if (n > size) {
            throw std::runtime_error("n is greater than size.");
        }
        if (n < 0 || m < 0 || n < m) {
            return 0L;
        }
        return table[n][m];
    }

private:
    int size;
    std::vector<std::vector<long>> table;
};

class ArrayGraph {
public:
    ArrayGraph(int n) : costArray(n, std::vector<std::optional<long>>(n)), vertexNum(n) {}

    void link(int from, int to, long cost) {
        costArray[from][to] = cost;
    }

    std::optional<long> getCost(int from, int to) {
        return costArray[from][to];
    }

    int getVertexNum() {
        return vertexNum;
    }

private:
    std::vector<std::vector<std::optional<long>>> costArray;
    int vertexNum;
};

class DfsFlowResolver {
public:
    DfsFlowResolver(ArrayGraph graph) : graph(graph) {}

    long maxFlow(int from, int to) {
        long sum = 0L;
        long currentFlow;
        do {
            currentFlow = flow(from, to, LONG_MAX / 3, std::vector<bool>(graph.getVertexNum()));
            sum += currentFlow;
        } while (currentFlow > 0);
        return sum;
    }

private:
    ArrayGraph graph;

    long flow(int from, int to, long current_flow, std::vector<bool>& passed) {
        passed[from] = true;
        if (from == to) {
            return current_flow;
        }
        for (int id = 0; id < graph.getVertexNum(); id++) {
            if (passed[id]) {
                continue;
            }
            std::optional<long> cost = graph.getCost(from, id);
            if (cost.value_or(0L) > 0) {
                long nextFlow = current_flow < cost.value() ? current_flow : cost.value();
                long returnFlow = flow(id, to, nextFlow, passed);
                if (returnFlow > 0) {
                    graph.link(from, id, cost.value() - returnFlow);
                    graph.link(id, from, graph.getCost(id, from).value_or(0L) + returnFlow);
                    return returnFlow;
                }
            }
        }
        return 0L;
    }
};

class BinaryIndexedTree {
public:
    BinaryIndexedTree(int size) : array(size + 1) {}

    void add(int index, long value) {
        for (int i = index; i < array.size(); i += (i & -i)) {
            array[i] += value;
        }
    }

    long getSum(int index) {
        long sum = 0L;
        for (int i = index; i > 0; i -= (i & -i)) {
            sum += array[i];
        }
        return sum;
    }

private:
    std::vector<long> array;
};

class BinaryIndexedTree2D {
public:
    BinaryIndexedTree2D(int size1, int size2) : array(size1 + 1) {
        for (int i = 1; i <= size1; i++) {
            array[i] = std::vector<long>(size2 + 1);
        }
    }

    void add(int index1, int index2, long value) {
        for (int i1 = index1; i1 < array.size(); i1 += (i1 & -i1)) {
            for (int i2 = index2; i2 < array[i1].size(); i2 += (i2 & -i2)) {
                array[i1][i2] += value;
            }
        }
    }

    long getSum(int index1, int index2) {
        long sum = 0L;
        for (int i1 = index1; i1 > 0; i1 -= (i1 & -i1)) {
            for (int i2 = index2; i2 > 0; i2 -= (i2 & -i2)) {
                sum += array[i1][i2];
            }
        }
        return sum;
    }

private:
    std::vector<std::vector<long>> array;
};

class ArrayUnionFind {
public:
    ArrayUnionFind(int size) : parent(size), rank(size), size(size) {
        for (int i = 0; i < size; i++) {
            parent[i] = i;
        }
    }

    void union(int A, int B) {
        int rootA = root(A);
        int rootB = root(B);
        if (rootA != rootB) {
            if (rank[rootA] < rank[rootB]) {
                unionTo(rootA, rootB);
            } else {
                unionTo(rootB, rootA);
                if (rank[rootA] == rank[rootB]) {
                    rank[rootA]++;
                }
            }
        }
    }

    bool judge(int A, int B) {
        return root(A) == root(B);
    }

    std::set<int> getSet(int id) {
        std::set<int> set;
        for (int i = 0; i < size; i++) {
            if (judge(i, id)) {
                set.insert(i);
            }
        }
        return set;
    }

protected:
    void unionTo(int source, int dest) {
        parent[source] = dest;
    }

    int root(int id) {
        if (parent[id] == id) {
            return id;
        }
        parent[id] = root(parent[id]);
        return parent[id];
    }

private:
    std::vector<int> parent;
    std::vector<int> rank;
    int size;
};

class SetUnionFind : public ArrayUnionFind {
public:
    SetUnionFind(int size) : ArrayUnionFind(size) {
        for (int i = 0; i < size; i++) {
            map[i].insert(i);
        }
    }

    std::set<int> getSet(int id) {
        return map[root(id)];
    }

protected:
    void unionTo(int source, int dest) {
        ArrayUnionFind::unionTo(source, dest);
        map[dest].insert(map[source].begin(), map[source].end());
    }

private:
    std::unordered_map<int, std::set<int>> map;
};

class PrimeNumberUtils {
public:
    PrimeNumberUtils(int limit) : limit(limit), isPrimeArray(limit), primes() {
        if (limit > 10000000) {
            std::cerr << "上限の値が高すぎるため素数ユーティリティの初期化でTLEする可能性が大変高いです";
        }
        if (limit > 2) {
            primes.push_back(2);
            isPrimeArray[2] = true;
        }

        for (int i = 3; i < limit; i += 2) {
            if (isPrime(i, primes)) {
                primes.push_back(i);
                isPrimeArray[i] = true;
            }
        }
    }

    std::vector<int> getPrimeNumberList() {
        return primes;
    }

    bool isPrime(int n) {
        return isPrimeArray[n];
    }

private:
    int limit;
    std::vector<bool> isPrimeArray;
    std::vector<int> primes;

    bool isPrime(int n, std::vector<int>& primes) {
        for (int prime : primes) {
            if (n % prime == 0) {
                return false;
            }
            if (prime > std::sqrt(n)) {
                break;
            }
        }
        return true;
    }
};

class LongBit {
public:
    LongBit(int size) : bitArray((size + 63) / 64) {}

    void set(int index, bool bit) {
        int segment = index / 64;
        int inIndex = index % 64;
        if (bit) {
            bitArray[segment] |= 1L << inIndex;
        } else {
            bitArray[segment] &= ~(1L << inIndex);
        }
    }

    bool get(int index) {
        int segment = index / 64;
        int inIndex = index % 64;
        return (bitArray[segment] & (1L << inIndex)) != 0L;
    }

    void shiftRight(int num) {
        int shiftSeg = num / 64;
        int shiftInI = num % 64;
        for (int segment = 0; segment < bitArray.size(); segment++) {
            int sourceSeg = segment + shiftSeg;
            if (sourceSeg < bitArray.size()) {
                bitArray[segment] = bitArray[sourceSeg] >> shiftInI;
                if (shiftInI > 0 && sourceSeg + 1 < bitArray.size()) {
                    bitArray[segment] |= bitArray[sourceSeg + 1] << (64 - shiftInI);
                }
            } else {
                bitArray[segment] = 0L;
            }
        }
    }

    void shiftLeft(int num) {
        int shiftSeg = num / 64;
        int shiftInI = num % 64;
        for (int segment = bitArray.size() - 1; segment >= 0; segment--) {
            int sourceSeg = segment - shiftSeg;
            if (sourceSeg >= 0) {
                bitArray[segment] = bitArray[sourceSeg] << shiftInI;
                if (shiftInI > 0 && sourceSeg > 0) {
                    bitArray[segment] |= bitArray[sourceSeg - 1] >> (64 - shiftInI);
                }
            } else {
                bitArray[segment] = 0L;
            }
        }
    }

    long getLong(int segment) {
        return bitArray[segment];
    }

    void or(LongBit& bitset) {
        for (int segment = 0; segment < bitArray.size(); segment++) {
            bitArray[segment] |= bitset.getLong(segment);
        }
    }

    void and(LongBit& bitset) {
        for (int segment = 0; segment < bitArray.size(); segment++) {
            bitArray[segment] &= bitset.getLong(segment);
        }
    }

private:
    std::vector<long> bitArray;
};

// End of Code
